src/bit_machine/mod.rs

```diff
@@ -187,9 +187,12 @@ impl BitMachine {
         if !input.clone().is_of_type(self.source_ty.clone()) {
             return Err(ExecutionError::InputWrongType(self.source_ty.clone()));
         }
-        self.new_frame(input.len());
-        self.write_value(&input);
-        self.move_frame();
+        // Unit value doesn't need extra frame
```

Comment:
In 440298b621bf0f9f0e68544fe7005494d0581891:

This optimization is fine but where is the panic that you claim exists in the commit message?

Reply:
The panic occurs during execution, a few lines down the code. The error message is "Program requires a non-empty input to execute". This error is triggered if there is a frame on the input stack (unit value) but the input width is zero (unit type). The check is made less confusing in 564fa2566b4a65705c29af0c03d20fa929188c5a.

Reply:
I think the assertion was just wrong. It is very weirdly structured. But in any case it is removed in 563fa25 as you say.